pennylane/fermi/conversion.py

```diff
@@ -104,6 +111,10 @@ def _(fermi_operator: FermiWord, ps=False, wire_map=None):
                 }
             )
 
+    for pw in qubit_operator:
+        if tol and abs(qml.math.imag(qubit_operator[pw])) <= tol:
```

Comment:
```suggestion
        if tol is None and abs(qml.math.imag(qubit_operator[pw])) <= tol:
```

Right now, when `tol` is `0`, this doesn't remove any imaginary components of 0j